src/pyaro/timeseries/Filter.py

```diff
@@ -1184,3 +1184,162 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
+        self._radius = radius
+        self._lower = lower
+        self._upper = upper
+        self._topography = None
+
+    @property
+    def topography(self):
+        if "cf_units" not in sys.modules:
+            raise ModuleNotFoundError(
+                "valleyfloorrelativealtitudefilter filter is missing required dependency 'cf-units'. Please install to use this filter."
+            )
+        if "xarray" not in sys.modules:
+            raise ModuleNotFoundError(
+                "valleyfloorrelativealtitudefilter filter is missing required dependency 'xarray'. Please install to use this filter."
+            )
+
+        if self._topography is None:
+            try:
+                with xr.open_dataset(self._topo_file) as topo:
+                    topo = topo.fillna(0)
+                    self._topography = topo
+            except Exception as ex:
+                raise FilterException(
+                    f"Cannot read topography from '{self._topo_file}:{self._topo_var}' : {ex}"
+                )
+        return self._topography
+
+    def init_kwargs(self):
+        return {
+            "topo_file": self._topo_file,
+            "topo_var": self._topo_var,
+            "radius": self._radius,
+            "lower": self._lower,
+            "upper": self._upper,
+        }
+
+    def name(self):
+        return "valleyfloorrelativealtitudefilter"
+
+    def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
+        filtered_stations = {}
+        for k, v in stations.items():
+            lat = v.latitude
+            lon = v.longitude
+            alt = v.altitude
+
+            ralt = self._calculate_relative_altitude(
+                lat, lon, radius=self._radius, altitude=alt
+            )
+
+            keep = True
+            if self._lower is not None:
+                if self._lower > ralt:
+                    keep = False
+            if self._upper is not None:
+                if self._upper < ralt:
+                    keep = False
+            if keep:
+                filtered_stations[k] = v
+
+        return filtered_stations
+
+    def _calculate_relative_altitude(
+        self, lat: float, lon: float, *, radius: float, altitude: float
+    ):
+        # At most one degree of latitude (at equator) is roughly 111km.
+        # Subsetting to based on this value with safety margin makes the
+        # distance calculation MUCH more efficient.
+        s = 0.1 + (radius / 1000) / 100
+        topo = self.topography.sel(
+            lon=slice(lon - s, lon + s), lat=slice(lat - s, lat + s)
+        )
+
+        distances = self._haversine(topo["lon"], topo["lat"], lon, lat)
+        within_radius = distances <= radius
+
+        values_within_radius = topo[self._topo_var].where(within_radius, drop=True)
+
+        min_value = float(values_within_radius.min())
+        return altitude - max([min_value, 0])
+
+    def _find_lat_lon_variables(self, topo_xr):
```

Comment:
Is this a copy of _find_lat_lon_variables() of the relative_altitude_filter?
Since the both filters are very similar, maybe they could inherit from the same base-class and share common functions?